src/status_im/models/wallet.cljs

```diff
@@ -157,24 +183,24 @@
           (dissoc :message-id :id :gas)))))
 
 (defn handle-transaction-error [{:keys [db]} {:keys [code message]}]
-  (let [{:keys [dapp-transaction]} (get-in db [:wallet :send-transaction])]
+  (let [{:keys [on-error]} (get-in db [:wallet :send-transaction])]
     (case code
 
       ;;WRONG PASSWORD
       constants/send-transaction-err-decrypt
       {:db (-> db
                (assoc-in [:wallet :send-transaction :wrong-password?] true))}
 
-      (cond-> (let [cofx {:db
-                          (-> db
-                              (assoc-in [:wallet :transactions-queue] nil)
-                              (assoc-in [:wallet :send-transaction] {}))
-                          :wallet/show-transaction-error
-                          message}]
-                (navigation/navigate-back cofx))
+      (merge (let [cofx {:db
+                         (-> db
+                             (assoc-in [:wallet :transactions-queue] nil)
+                             (assoc-in [:wallet :send-transaction] {}))
+                         :wallet/show-transaction-error
+                         message}]
+               (navigation/navigate-back cofx))
 
-        dapp-transaction
-        (web3-error-callback db dapp-transaction message)))))
+             (when on-error
+               {:dispatch (conj on-error "discarded")})))))
```

Comment:
What's "discarded"?

Reply:
we need some error message when a user cancelled a transaction, "transaction was cancelled by user" ?